agent/engine/common_integ_test.go

```diff
@@ -117,6 +117,17 @@ func verifyContainerRunningStateChangeWithRuntimeID(t *testing.T, taskEngine Tas
 		"Expected container runtimeID should not empty")
 }
 
+func verifyExecAgentRunningStateChange(t *testing.T, taskEngine TaskEngine) {
+	stateChangeEvents := taskEngine.StateChangeEvents()
+	event := <-stateChangeEvents
+	containerEvent := event.(api.ContainerStateChange)
+	assert.NotEmpty(t, containerEvent.ManagedAgents, "Expeceted exec-agent event has no managed agents")
+	if containerEvent.ManagedAgents != nil {
+		assert.Equal(t, apicontainerstatus.ManagedAgentRunning, containerEvent.ManagedAgents[0].Status,
```

Comment:
just curious, do we need to check for other indexes like `1,2,...`? or should we assert that `containerEvent.ManagedAgents` only has 1 element?

Reply:
We'll only have one message here.